WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Categories/ProductCategoryViewModelBuilderTests.swift

```diff
@@ -0,0 +1,136 @@
+
+        let rootViewModels = viewModels.filter { rootCategoriesNames.contains($0.name) }
+        XCTAssertEqual(rootViewModels.count, rootCategories.count)
+        rootViewModels.forEach {
+            XCTAssertEqual($0.indentationLevel, 0)
+        }
+
+        let subViewModels = viewModels.filter { subCategories1Names.contains($0.name) }
+        XCTAssertEqual(subViewModels.count, subCategories1.count)
+        subViewModels.forEach {
+            XCTAssertEqual($0.indentationLevel, 1)
+        }
+
+        let subViewModels2 = viewModels.filter { subCategories2Names.contains($0.name) }
+        XCTAssertEqual(subViewModels2.count, subCategories2.count)
+        subViewModels2.forEach {
+            XCTAssertEqual($0.indentationLevel, 2)
+        }
+    }
+
+    func testRootCategoriesAreSelectedWhenProvidingProductSelectedCategories() {
+        // Given
+        let categories = sampleCategories(initialID: 1, count: 10)
+        let selectedCategories = sampleCategories(initialID: 3, count: 5)
+
+        // When
+        let viewModels = ProductCategoryListViewModel.CellViewModelBuilder.viewModels(from: categories, selectedCategories: selectedCategories)
+
+        // Then
+        let selectedCategoryNames = selectedCategories.map { $0.name }
+        let selectedViewModelsNames = viewModels.filter { $0.isSelected }.map { $0.name }
+        XCTAssertEqual(selectedCategoryNames, selectedViewModelsNames)
+    }
+
+    func testSubCategoriesAreSelectedWhenProvidingProductSelectedCategoriesWithSubCategories() {
+        // Given
+        let rootCategories = sampleCategories(initialID: 1, count: 10)
+        let subCategories1 = sampleCategories(initialID: 11, parentID: 5, count: 5)
+        let selectedCategories = sampleCategories(initialID: 12, parentID: 5, count: 2)
+        let allCategories = rootCategories + subCategories1
+
+        // Then
+        let viewModels = ProductCategoryListViewModel.CellViewModelBuilder.viewModels(from: allCategories, selectedCategories: selectedCategories)
+
+        // When
+        let selectedCategoryNames = selectedCategories.map { $0.name }
+        let selectedViewModelsNames = viewModels.filter { $0.isSelected }.map { $0.name }
+        XCTAssertEqual(selectedCategoryNames, selectedViewModelsNames)
+    }
+
+    func testViewModelGenerationTimeWithFromBigSetOfCategories() {
+        let rootCategories = sampleCategories(initialID: 1, count: 1000)
+        let subCategories1 = sampleCategories(initialID: 1001, parentID: 5, count: 500)
+        let subCategories2 = sampleCategories(initialID: 1601, parentID: 13, count: 300)
+        let selectedCategories = sampleCategories(initialID: 1100, parentID: 5, count: 100)
+        let allCategories = rootCategories + subCategories1 + subCategories2
+        measure {
+            _ = ProductCategoryListViewModel.CellViewModelBuilder.viewModels(from: allCategories, selectedCategories: selectedCategories)
+        }
```

Comment:
Hmm. This is isn't how I expected it to work. I guess the baselines have are specific to the machines they're used on. I thought we could share them. 😞 

![image](https://user-images.githubusercontent.com/198826/79502067-54cdbc00-7fec-11ea-835c-8d6771b03014.png)

Anyway, let's just keep this for now.